### ISSUE TITLE
[shapes] DrawPixel vs DrawLine seems to render at different pixel locations.

### ISSUE THREAD

**Issue by lpalozzi** [raysan5/raylib#1863]
### Issue description

DrawPixel vs DrawLine seems to render at different pixel locations.

### Environment

Debian Linux, Opengl 3.3, NVIDIA GPU

### Issue Screenshot
![image](https://user-images.githubusercontent.com/54338305/124615838-75b94180-de2a-11eb-97d0-d40cdb9ad6f3.png)

### Code Example
```
void draw()
{
    Vector2 p1 = (Vector2){100, 50};
    Vector2 p2 = (Vector2){100, 150};
    Vector2 p3 = (Vector2){50, 100};
    Vector2 p4 = (Vector2){150, 100};

    DrawLineV(p1, p2, GREEN);
    DrawLineV(p3, p4, GREEN);
    DrawPixelV(p1, WHITE);
    DrawPixelV(p2, WHITE);
    DrawPixelV(p3, WHITE);
    DrawPixelV(p4, WHITE);
}
```

**Comment by raysan5** [comment]
This issue is really tricky, it depends on multiple factors:

 1. GPU and the drivers. For example here it is the result testing your code on my Intel HD Graphics: 

![screenshot000](https://user-images.githubusercontent.com/5766837/124636933-64535200-de89-11eb-96f9-9d560afdda61.png)

 2. If you check the pixels positions, in my case, they match the expected defined position but they can have an offset of 1 pixel, again, depending on the GPU and the drivers. So, you can think the problem is on the line drawing... we are in the same situation:

 3. Rasterizer. when the mathematical defined world is discretized into a pixel-based image, GPU (and the driver) decides how rasterization happens.

 4. As a small latest detail, raylib draws points using 1px lines... but that's just an implementation detail.

In conclusion, you can review `DrawLineV()` and `DrawPixelV()` functions in `shapes.c` module to fit perfectly your needs but result could vary depending the GPU (and drivers) where your program is executed.

**Comment by McUsr** [comment]
Nice to know when you play with line drawing algorithms in Raylib, drawing one pixel at a time. :)
